NSubstitute.DbConnection/NSubstitute.DbConnection/MockQuery.cs

```diff
@@ -148,6 +148,16 @@ public DbDataReader ExecuteReader(IDbCommand mockCommand)
             mockReader[Arg.Any<int>()].Returns(ci => GetValue(FieldByIndex((int)ci[0])));
             mockReader.GetValue(Arg.Any<int>()).Returns(ci => GetValue(FieldByIndex((int)ci[0])));
             mockReader[Arg.Any<string>()].Returns(ci => GetValue(FieldByName((string)ci[0])));
+            mockReader.GetValues(Arg.Any<object[]>()).Returns(ci =>
+            {
+                var values = ci.ArgAt<object[]>(0);
+                for (var fieldIndex = 0; fieldIndex < values.Length; fieldIndex++)
```

Comment:
Looks like this would fail out of range for certain wrongly-sized arrays. I'd be happy to assert the array needs to be the correct size here and just fail with a more descriptive error (i.e. be opinionated here rather than try to mock the edge cases), or otherwise we could handle and have test cases for

1. array the same width as the result set
2. array narrower than the result set
3. array wider that the result set

Reply:
I used ODBC [Reference Source](https://referencesource.microsoft.com/#System.Data/fx/src/data/System/Data/Odbc/OdbcDataReader.cs,502) as the model to properly handle this scenario.